### PR TITLE
Adjust plans for pricing formulas

### PR DESCRIPTION
**Pull request by adamaltman**

### CODE REVIEW

File: spec/definitions/Plan.yaml [hunk 1]
```diff
@@ -45,10 +51,50 @@ properties:
...
+      properties:
+        price:
+          description: If the price is 0, it's free.
+          type: number
+          format: double
+          example: 99.95
+        maxQuantity:
+          description: |
+            If the value is `null`, it's assumed to be infinite.  It starts at the end of the previous bracket.
+            If there is no previous bracket, it starts at 1.
```
Comment (colinbird) [inline spec/definitions/Plan.yaml]: This is a bit confusing but I think I understand it, maybe an example would help

File: spec/definitions/Plans/Plan.yaml [hunk 2]
```diff
@@ -19,50 +20,55 @@ properties:
...
-  trialPeriodLength:
-    description: The length of time (used with the trialPeriodUnit)
-    type: integer
-  setupAmount:
-    description: The amount of a setup - 0 is a valid value (for free)
-    type: number
-    format: double
+  pricing:
+    $ref: "#/definitions/PlanPriceFormula"
+  recurringInterval:
```
Comment (2e3s) [inline spec/definitions/Plans/Plan.yaml]: It is inconsistent. In trial it is called "period", here "interval".

File: spec/definitions/Plans/Plan.yaml [hunk 3]
```diff
@@ -19,50 +20,55 @@ properties:
...
+  trial:
+    type: object
+    description: The trial. Set `null` if no trial.
+    required:
+      - price
+      - period
+    properties:
+      price:
+        description: The price of the trial. For a free trial, use `0`.
+        type: number
```
Comment (2e3s) [inline spec/definitions/Plans/Plan.yaml]: Will the trial be priced in a complex way?

**Comment by 2e3s** [comment]
Would there be a field to denote a variable usage?

File: spec/definitions/Plans/Plan.yaml (hunk 2 above)
Comment (slavcodev) [reply spec/definitions/Plans/Plan.yaml]: Well, it's not supposed to be consistent.
Trial - is a period of time when product is provided for special price.
The main price is billed in intervals, isn't it?

File: spec/definitions/Plans/Plan.yaml (hunk 3 above)
Comment (slavcodev) [reply spec/definitions/Plans/Plan.yaml]: I do not know. Should it be? We've not discussed such requirement.

**Comment by slavcodev** [comment]
> Would there be a field to denote a variable usage?

Yes. But the changes related to the `Usage AP` will be in different PR, just to not block the release with plan pricing.

File: spec/definitions/Plans/Plan.yaml (hunk 2 above)
Comment (2e3s) [reply spec/definitions/Plans/Plan.yaml]: To me "period" and "interval" are just synonyms

File: spec/definitions/Plans/Plan.yaml (hunk 3 above)
Comment (2e3s) [reply spec/definitions/Plans/Plan.yaml]: I don't know. I can imagine a trial not being flat rate but just reduced recurring price (which may be complex). Probably this should be discussed before going forward.

File: spec/definitions/Plans/Plan.yaml (hunk 2 above)
Comment (slavcodev) [reply spec/definitions/Plans/Plan.yaml]: are they?

Comment (2e3s) [reply spec/definitions/Plans/Plan.yaml]: More strictly saying, a period is an interval of time.

Comment (2e3s) [reply spec/definitions/Plans/Plan.yaml]: To me it's okay to name `recurring` by analog of trial

Comment (2e3s) [reply spec/definitions/Plans/Plan.yaml]: And maybe have `pricing` with this object?

Comment (slavcodev) [reply spec/definitions/Plans/Plan.yaml]: I do not think so. For me `recurring interval` and `pricing` are two different things, for example you can have `price` w/o specified `recurring interval`

File: spec/definitions/Plans/Plan.yaml (hunk 3 above)
Comment (adamaltman) [reply spec/definitions/Plans/Plan.yaml]: No.

**Comment by slavcodev** [comment]
devs please review

**Pull request merged**